src/lib/Preview.js

```diff
@@ -674,7 +676,10 @@ class Preview extends EventEmitter {
             }
             /* eslint-enable camelcase */
         } else {
-            throw new Error(
+            throw new PreviewError(
+                ERROR_CODE.BAD_INPUT,
+                __('error_generic'),
+                {},
```

Comment:
I'm throwing errors when possible as to not call `triggerError()` manually too many times.